src/BaGet.Core/Search/DatabaseSearchService.cs

```diff
@@ -178,32 +178,39 @@ public Task IndexAsync(Package package, CancellationToken cancellationToken)
             var frameworks = GetCompatibleFrameworksOrNull(framework);
             var search = (IQueryable<Package>)_context.Packages.Where(p => p.Listed);
 
-            if (!string.IsNullOrEmpty(query))
+            IQueryable<Package> ComposePackageQuery(IQueryable<Package> packageQuery)
```

Comment:
What do you think of naming this `AddSearchFilters`?

```suggestion
            IQueryable<Package> AddSearchFilters(IQueryable<Package> packageQuery)
```

Reply:
Nice idea using a local function, it's a very clean solution 👍